server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -356,7 +353,11 @@ private void waitDecommissionFinish() {
     while (isDecommissioning()) {
       remainApplicationNum = shuffleTaskManager.getAppIds().size();
       if (remainApplicationNum == 0) {
-        serverStatus.set(ServerStatus.DECOMMISSIONED);
```

Comment:
State could change between `isDecommissioning()` and `serverStatus.set(ServerStatus.DECOMMISSIONED)`.